lib/posix/stropts.h

```diff
@@ -0,0 +1,135 @@
+#define I_RECVFD    /* Get a file descriptor.*/
+#define I_SENDFD    /* Pass a file descriptor.*/
+#define I_SETCLTIME /* Set close time delay.*/
+#define I_SETSIG    /* Ask for notification signals.*/
+#define I_SRDOPT    /* Set the read mode.*/
+#define I_STR       /* Send a STREAMS ioctl().*/
+#define I_SWROPT    /* Set the write mode.*/
+#define I_UNLINK    /* Disconnect STREAMs.*/
+
+#define FMNAMESZ /* Buffer size for I_LOOK.*/
+
+#define FLUSHR  /* Flush read queues.*/
+#define FLUSHRW /* Flush read and write queues.*/
+#define FLUSHW  /* Flush write queues.*/
+
+#define S_BANDURG /* Generate SIGURG for priority messages.*/
+#define S_ERROR   /* Notification of error condition.*/
+#define S_HANGUP  /* Notification of a hangup.*/
+#define S_HIPRI   /* High-priority message present.*/
+#define S_INPUT   /* Message arrived at STREAM head.*/
+#define S_MSG     /* STREAMS signal message received.*/
+#define S_OUTPUT  /* Write queue is not full.*/
+#define S_RDBAND  /* Non-zero priority message arrived.*/
+#define S_RDNORM  /* Normal message arrived.*/
+#define S_WRBAND  /* Non-zero priority write queue not full.*/
+#define S_WRNORM  /* Equivalent to S_OUTPUT.*/
+
+#define RS_HIPRI /* Look only for high-priority messages.*/
+
+#define RMSGD     /* Message-discard mode.*/
+#define RMSGN     /* Message-non-discard mode.*/
+#define RNORM     /* Byte-STREAM mode, default.*/
+#define RPROTDAT  /* Deliver control part as data.*/
+#define RPROTDIS  /* Discard control part, deliver data.*/
+#define RPROTNORM /* Fail read if control part present.*/
+
+#define SNDZERO /* Send zero-length message on 0 byte write.*/
+
+#define ANYMARK  /* Check if the message is marked.*/
+#define LASTMARK /* Check if it's the last marked message.*/
+
+#define MUXID_ALL /* Unlink all STREAMs linked to the file descriptor.*/
+
+#define MORECTL   /* More control information in message.*/
+#define MOREDATA  /* More data left in message.*/
+#define MSG_ANY   /* Receive any message.*/
+#define MSG_BAND  /* Receive message from specified band.*/
+#define MSG_HIPRI /* Send/receive high-priority message.*/
+
+typedef int t_scalar_t;
+typedef unsigned int t_uscalar_t;
+
+struct bandinfo {
+	int bi_flag;
+	unsigned char bi_pri;
+};
+
+struct strpeek {
+	struct strbuf ctlbuf;
+	struct strbuf databuf;
+	t_uscalar_t flags;
+};
+
+struct strbuf {
+	int maxlen;
+	int len;
+	char *buf;
+};
+
+struct strfdinsert {
+	struct strbuf ctlbuf;
+	struct strbuf databuf;
+	t_uscalar_t flags;
+	int fildes;
+	int offset;
+};
+
+struct strioctl {
+	int ic_cmd;
+	char *ic_dp;
+	int ic_len;
+	int ic_timout;
+};
+
+struct strrecvfd {
+	int fd;
+	uid_t uid;
+	gid_t gid;
+};
+
+struct str_list {
+	struct str_mlist *sl_modlist;
+	int sl_nmods;
+};
+
+struct str_mlist {
+	char l_name[FMNAMESZ + 1];
+	struct str_mlist *l_next;
+};
+
+int fattach(int, const char *);
+int fdetach(const char *);
+int getmsg(int, struct strbuf *restrict buf1, struct strbuf *restrict buf2, int *restrict flags);
```

Comment:
Zephyr uses `ZRESTRICT` for these

```suggestion
int getmsg(int, struct strbuf *ZRESTRICT buf1, struct strbuf *ZRESTRICT buf2, int *ZRESTRICT flags);
```